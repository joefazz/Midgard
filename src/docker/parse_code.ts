import { Repl } from "../types";

export function getCodeSaveCommand(filename: string, code: string) {
    let cmd = ["/bin/bash", "-c"];

    // Without this line the echo will break if the user decides they want to use a string
    code = code.replace(/"/g, `\\"`);
    code = code.replace(/`/g, "\\`");

    cmd.push(`echo "${code}" > ${filename}`);

    return cmd;
}

export function getCodeExecutionCommand(repl: Repl, file: string) {
    let command = ["/bin/bash", "-c"];

    if (repl === Repl.C) {
        return command.concat(`gcc ${file} && ./a.out && rm a.out`);
    } else {
        return command.concat(`${repl} ${file}`);
    }
}
