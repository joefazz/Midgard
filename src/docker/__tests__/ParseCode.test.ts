import { getCodeSaveCommand } from "../parse_code";

describe("Test Code Parsing Functions", () => {
    const MOCK_CODE = "let x = 7;console.log(x*x);";
    const MOCK_FILE = "testfile.js";

    test("Code save command", () => {
        expect(getCodeSaveCommand(MOCK_FILE, MOCK_CODE)).toMatchSnapshot();
    });
});
