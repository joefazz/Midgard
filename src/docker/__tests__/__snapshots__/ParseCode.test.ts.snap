// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Code Parsing Functions Code save command 1`] = `
Array [
  "/bin/bash",
  "-c",
  "echo \\"let x = 7;console.log(x*x);\\" > testfile.js",
]
`;
